sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/WalkedTypePathRecorder.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst
+
+object WalkedTypePathRecorder {
```

Comment:
I was thinking about making this as class (with storing built path in each instance) but soon realized it requires touching other thing as well and feel a bit overkill. I'm still open to make this as individual class so please let me know if it sounds better.

Reply:
making it a class looks better, as it needs to accumulate the walked type path.

Reply:
Thanks for the support! Just addressed.